recipes/filepattern/build-lib.sh

```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
+
+set -eux # Abort on error.
+
+mkdir build-lib
+cd build-lib
+
+if [[ "$OSTYPE" == "darwin"* ]]; then
+	export CXXFLAGS="${CXXFLAGS} -D_LIBCPP_DISABLE_AVAILABILITY"
+fi
+
+cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_PREFIX_PATH=${PREFIX} -DCMAKE_INSTALL_PREFIX=${PREFIX} -Dfilepattern_SHARED_LIB=ON -DRUN_GTEST=OFF  ..
+
+cmake --build . --target install --parallel
```

Comment:
Is there a `ctest`, `make check` or similar that we can run here?

Reply:
We are doing some install check that are described in `meta.yaml`. Will that be sufficient?
Naive question: What will be the purpose of `ctest` or `make check`?

Reply:
If they exists, test the library. Do they? If not there isn't anything we can do here. I asked b/c most of the time compiled libraries do have them.

Reply:
We can change the `DRUN_GTEST=OFF` to `DRUN_GTEST=ON` and then run the tests.

Reply:
also, we have this following section in `meta.yaml` to check whether the files are installed correctly.
```
    test:
      commands:
        - test -f $PREFIX/include/filepattern.h  # [unix]
        - if not exist %LIBRARY_INC%\\filepattern.h exit 1  # [win]
        - test -f $PREFIX/lib/libfilepattern.so.2  # [linux]
        - test -f $PREFIX/lib/libfilepattern.2.dylib  # [mac]
        - if not exist %LIBRARY_BIN%\\filepattern.dll exit 1  # [win]
        - test -f $PREFIX/lib/cmake/filepattern/filepatternConfig.cmake  # [unix]
        - if not exist %LIBRARY_LIB%\\cmake\\filepattern\\filepatternConfig.cmake exit 1  # [win]
```

Reply:
Existence checks are typically sufficient. We don't need to run the entire CI suite for every package. We do need to check that all the artifacts are installed in the correct place, that link paths are set correctly, and that the ABI won't break.

Reply:
These checks should be sufficient then since they make sure all the artifacts are installed in correct location.